include/nanobind/stl/chrono.h

```diff
@@ -99,7 +99,13 @@ template <typename type> class duration_caster {
         return pack_timedelta(dd.count(), ss.count(), us.count());
     }
 
-    NB_TYPE_CASTER(type, const_name("datetime.timedelta"))
+    #if PY_VERSION_HEX < 0x03090000
+        NB_TYPE_CASTER(type, io_name("typing.Union[datetime.timedelta, float]",
+                                     "datetime.timedelta"))
```

Comment:
in stubs, it's fine to use the modern syntax all the time. 
Typecheckers treat stubs as all having `from __future__ import annotations`

Reply:
In the past, there were users who wanted old-style annotations when working with old Python versions, and to import various things from `typing_extensions` that potentially don't exist on them (even though it's just the stubs!) 🤷.

Reply:
The rest of nanobind's type casters use this form (`typing.Union`), so I'm going to stay consistent with that.